### PR TITLE
ID-42

### PR DESCRIPTION
**Pull request by SharmaDinesh**

### CODE REVIEW

File: app/screens/claim/index.tsx [hunk 1]
```diff
@@ -29,9 +30,16 @@ const Claim = () => {
...
-          await AsyncStorage.setItem("document_url", result.uri);
+          const hashKey = await Crypto.digestStringAsync(
+            Crypto.CryptoDigestAlgorithm.SHA256,
+            result.uri
+          );
+          const data = {
+            'hash': hashKey,
+            'base64Url': result.uri
+          }
+          await AsyncStorage.setItem("document_url", JSON.stringify(data));
```
Comment (rellfy) [inline app/screens/claim/index.tsx:42]:
```suggestion
          await AsyncStorage.setItem("document_url", JSON.stringify(await getData(result)));
```

File: app/screens/claim/index.tsx [hunk 2]
```diff
@@ -78,7 +94,15 @@ const Claim = () => {
...
-        await AsyncStorage.setItem("camera_url", result.uri);
+        const hashKey = await Crypto.digestStringAsync(
+          Crypto.CryptoDigestAlgorithm.SHA256,
+          result.uri
+        );
+        const data = {
+          'hash': hashKey,
+          'base64Url': result.uri
+        }
+        await AsyncStorage.setItem("camera_url", JSON.stringify(data));
```
Comment (rellfy) [inline app/screens/claim/index.tsx:105]:
```suggestion
        await AsyncStorage.setItem("camera_url", JSON.stringify(await getData(result)));
```

**Review by rellfy — COMMENTED** [review]
Just a suggestion to reduce the duplicate code. Might also be worth adding a husky hook at some point so it's linted on each commit as the prettier config is set to use single quotes.

File: app/screens/claim/index.tsx [hunk 3]
```diff
@@ -55,7 +63,15 @@ const Claim = () => {
...
-          await AsyncStorage.setItem("library_url", result.uri);
+          const hashKey = await Crypto.digestStringAsync(
+            Crypto.CryptoDigestAlgorithm.SHA256,
+            result.uri
+          );
+          const data = {
+            'hash': hashKey,
+            'base64Url': result.uri
+          }
+          await AsyncStorage.setItem("library_url", JSON.stringify(data));
```
Comment (rellfy) [inline app/screens/claim/index.tsx:74]:
```suggestion
          await AsyncStorage.setItem("library_url", JSON.stringify(await getData(result)));
```

File: app/screens/claim/index.tsx [hunk 4]
```diff
@@ -10,6 +10,7 @@ import MobileClaim from "./MobileClaim";
 import SelectClaim from "./SelectClaim";
 import * as ImagePicker from "expo-image-picker";
 import * as DocumentPicker from "expo-document-picker";
+import * as Crypto from 'expo-crypto';
```
[the remark this answers is not in the document]
Comment (SharmaDinesh) [reply app/screens/claim/index.tsx]: Done

File: app/screens/claim/index.tsx [hunk 5]
```diff
@@ -10,6 +10,7 @@ import MobileClaim from "./MobileClaim";
...
 import AsyncStorage from "@react-native-async-storage/async-storage";
```
[the remark this answers is not in the document]
Comment (SharmaDinesh) [reply app/screens/claim/index.tsx:15]: Done

File: app/screens/claim/index.tsx (hunk 1 above)
Comment (SharmaDinesh) [reply app/screens/claim/index.tsx]: Done

**Review by rellfy — APPROVED** [review]

**Review by bitcoinbrisbane — APPROVED** [review]

**Pull request merged**